### DISCUSSION TITLE
Is possible to specify a host to be resolved through a server

### DISCUSSION THREAD

**Discussion by Anivie** [DNSCrypt/dnscrypt-proxy#2458] [Q&A]
In some regions, some services are prohibited, but the correct address obtained through external dns resolution can be accessed normally. Therefore, I hope that all requests under a host name will be resolved by a certain dns server. Is there a similar configuration item that can be used?

**Comment by o101010** [comment]
Hi,

If I understand your message, you are speaking about hosts blocking by your ISP provider via DNS. They do that by settings their own DNS server in default DHCP request of ISP box/router.
By setting another DNS resolver, what does DNSCrypt-proxy, you are not limited anymore by your ISP DNS, because you don't use it anymore.

EDIT : If really needed, you can use [Forwarding](https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Forwarding) feature. Another discussion speak about those use case : [1880](https://github.com/DNSCrypt/dnscrypt-proxy/discussions/1880)

**Comment by Anivie** [comment]
> Hi,
> 
> If I understand your message, you are speaking about hosts blocking by your ISP provider via DNS. They do that by settings their own DNS server in default DHCP request of ISP box/router. By setting another DNS resolver, what does DNSCrypt-proxy, you are not limited anymore by your ISP DNS, because you don't use it anymore.
> 
> EDIT : If really needed, you can use [Forwarding](https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Forwarding) feature. Another discussion speak about those use case : [1880](https://github.com/DNSCrypt/dnscrypt-proxy/discussions/1880)

Thank you for your reply.

You may have misunderstood what I mean. In fact, my mean is: for example, when the Microsoft's New Bing service detects that the user is in China (through ip or other settings), it will be refused to access New Bing even if it has obtained account authorization.
The problem I am currently facing is: I have set up dns servers in China in the DNSCrypt-proxy, which will cause bing.com to be resolved to servers in China, after that, my proxy service will no longer work, thus I cannot access New Bing.
I hope that through some settings, the domain name related to bing.com will be forcibly resolved by google dns provider to connect to the server in the external area.

**Comment by lifenjoiner** [comment]
> cause bing.com to be resolved to servers in China, after that, my proxy service will no longer work

Although I have solutions to specify a host to be resolved through a specified server, you'd better change your config to proxy requests by host rather than by IP. That will let the host name to be resolved on the proxy server, and gets a better IP for the proxy server.
